sharktank/sharktank/layers/mmdit.py

```diff
@@ -144,3 +144,46 @@ def forward(
         txt = txt + txt_mod2.gate * txt_mlp_out3
 
         return img, txt
+
+
+class MMDITSingleBlock(ThetaLayer):
+    def __init__(self, theta, num_heads: int):
+        super().__init__(theta)
+
+        self.num_heads = num_heads
+        self.add_module("mod", ModulationLayer(theta("mod"), double=False))
+        self.add_module(
+            "attn_norm_q", RMSNormLayer(theta("attn.norm.query_norm"), epsilon=1e-6)
+        )
+        self.add_module(
+            "attn_norm_k", RMSNormLayer(theta("attn.norm.key_norm"), epsilon=1e-6)
+        )
+        self.add_module("attn_proj", LinearLayer(theta("attn.proj")))
+
+        self.add_module("linear1", LinearLayer(theta("linear1")))
+        self.add_module("linear2", LinearLayer(theta("linear2")))
+        self.hidden_size = 3072  # todo
+        self.mlp_hidden_dim = 3072  # todo
```

Comment:
Did you mean to leave these like that?

Reply:
Fleshed out the TODO, thanks for reminding me.